partiql-conformance-test-generator/src/generator.rs

```diff
@@ -79,63 +81,123 @@ impl Generator {
+        self.seen_fns.push(HashSet::new());
+        self.gen_variants(module.scope(), &namespace.contents);
+        self.seen_fns.pop();
     }
-}
 
-fn gen_test(scope: &mut Scope, test_case: &TestCase) {
-    let test_fn: &mut Function = scope.new_fn(&test_case.test_name.escape_test_name());
-    test_fn.attr("test");
-    for assertion in &test_case.assertions {
-        match assertion {
-            Assertion::SyntaxSuccess => {
-                test_fn.line(format!(
-                    r####"crate::pass_syntax(r#"{}"#);"####,
-                    &test_case.statement
-                ));
-            }
-            Assertion::SyntaxFail => {
-                test_fn.line(format!(
-                    r####"crate::fail_syntax(r#"{}"#);"####,
-                    &test_case.statement
-                ));
-            }
-            Assertion::NotYetImplemented => {
-                // for `NotYetImplemented` assertions, add the 'ignore' annotation to the test case
-                test_fn.attr("ignore = \"not yet implemented\"");
-                test_fn.attr("allow(unused_variables)");
+    fn intern_test_name(&mut self, mut name: String) -> String {
+        let seen_fns = self.seen_fns.last_mut().unwrap();
+
+        while seen_fns.contains(&name) {
+            name.push('_');
+        }
+
+        seen_fns.insert(name.clone());
+        name
+    }
+
+    fn gen_test(&mut self, scope: &mut Scope, test_case: &TestCase) {
+        let escaped_name = test_case.name.escape_test_name();
+        let name = self.intern_test_name(escaped_name);
+
+        let test_fn: &mut Function = scope.new_fn(&name);
+        test_fn.attr("test");
+        test_fn.attr("allow(text_direction_codepoint_in_literal)");
+
+        let doc = format!("Generated test for test named `{}`", &test_case.name);
+        test_fn.doc(&doc);
+
+        let mut ignore_test = false;
+
+        for assertion in &test_case.assert {
+            match assertion {
+                Assertion::SyntaxSuccess(_) => {
+                    test_fn.line(format!(
+                        r####"crate::pass_syntax(r#"{}"#);"####,
+                        &test_case.statement
+                    ));
+                }
+                Assertion::SyntaxFail(_) => {
+                    test_fn.line(format!(
+                        r####"crate::fail_syntax(r#"{}"#);"####,
+                        &test_case.statement
+                    ));
+                }
+                Assertion::StaticAnalysisFail(_) => {
+                    // TODO semantics tests are not yet implemented
+                    ignore_test = true;
+
+                    test_fn.line(format!(
+                        r####"crate::fail_semantics(r#"{}"#);"####,
+                        &test_case.statement
+                    ));
+                }
+                Assertion::EvaluationSuccess(_) => {
+                    // TODO semantics tests are not yet implemented
```

Comment:
nit: (probably will be addressed in upcoming PR)

```suggestion
                    // TODO evaluation success tests are not yet implemented
```